open_in_mpv/open_in_mpv.py

```diff
@@ -133,53 +173,39 @@ def real_main(log: TextIO) -> int:
     message. Then the message is expected to be proceed.
     """
     os.makedirs(dirname(MPV_SOCKET), exist_ok=True)
-    stdin_buffer = sys.stdin.buffer
-    req_len = struct.unpack('@i', stdin_buffer.read(4))[0]
-    message = json.loads(stdin_buffer.read(req_len).decode())
-    logger.debug(f'Message contents ({req_len}): {message}')
-    if 'init' in message:
-        resp = json.dumps(dict(version=VERSION, logPath=log.name, socketPath=MPV_SOCKET)).encode()
-        size = struct.pack('@i', len(resp))
-        stdout_buffer = sys.stdout.buffer
-        stdout_buffer.write(size)
-        stdout_buffer.write(resp)
+    message: Dict[str, Any] = request(sys.stdin.buffer)
+
+    if message['init']:
+        response(dict(version=VERSION, logPath=log.name, socketPath=MPV_SOCKET))
         log.close()
         return 0
-    try:
-        url: str = message['url']
-    except KeyError:
+
+    if ((url := message.get('url', None)) is None): # pylint: disable=C0325
         logger.exception('No URL was given')
         print(json.dumps(dict(message='Missing URL!')))
         return 1
+
     if (is_debug := message.get('debug', False)):
         logger.info('Debug mode enabled.')
+
     single: bool = message.get('single', True)
+
     # MacPorts
-    new_env = os.environ.copy()
-    data_resp: dict[str, Any] = dict(version=VERSION, log_path=log.name, message='About to spawn')
-    if isdir('/opt/local/bin'):
-        logger.info('Detected MacPorts. Setting PATH.')
-        data_resp['macports'] = True
-        old_path = os.environ.get('PATH')
-        new_env['PATH'] = '/opt/local/bin' if not old_path else ':'.join(
-            ('/opt/local/bin', old_path))
-    data_resp['env'] = new_env
-    if is_debug:
-        logger.debug('Environment:')
-        for k, value in new_env.items():
-            logger.debug(f'  {k}={value}')
+    data_resp: Dict[str, Any] = dict(version=VERSION, log_path=log.name, message='About to spawn')
+    data_resp['env'] = environment(data_resp, is_debug)
     logger.debug('About to spawn')
-    resp = json.dumps(data_resp).encode()
-    size = struct.pack('@i', len(resp))
-    stdout_buffer = sys.stdout.buffer
-    stdout_buffer.write(size)
-    stdout_buffer.write(resp)
+    response(data_resp)
+
     if exists(MPV_SOCKET) and single:
-        spawn(get_callback(url, log, new_env, is_debug))
+        spawn(get_callback(url, log, data_resp['env'], is_debug))
     else:
-        spawn_init(url, log, new_env, is_debug)
+        spawn_init(url, log, data_resp['env'], is_debug)
     logger.debug('mpv should open soon')
     logger.debug('Exiting with status 0')
+    if FALLBACKS['log'] is not None:
+        FALLBACKS['log'].cleanup()
+    if FALLBACKS['socket'] is not None:
+        FALLBACKS['socket'].close()
```

Comment:
Checking just boolean is sufficient for these conditionals. `if FALLBACKS['log']:`